query/query.go

```diff
@@ -0,0 +1,172 @@
+		}, "unable to parse the query fields")
+		return err
+	}
+	err = r.db.Create(q).Error
+	if err != nil {
+		// combination of title, space ID and creator should be unique
+		if gormsupport.IsUniqueViolation(err, "queries_title_space_id_creator_unique") {
+			log.Error(ctx, map[string]interface{}{
+				"err":      err,
+				"title":    q.Title,
+				"space_id": q.SpaceID,
+			}, "unable to create query because a query with same title already exists in the space by same creator")
+			return errors.NewDataConflictError(fmt.Sprintf("query already exists with title = %s , space_id = %s, creator = %s", q.Title, q.SpaceID, q.Creator))
+		}
+		log.Error(ctx, map[string]interface{}{}, "error adding Query: %s", err.Error())
+		return err
+	}
+	return nil
+}
+
+// List all queries in a space
+func (r *GormQueryRepository) List(ctx context.Context, spaceID uuid.UUID) ([]Query, error) {
+	defer goa.MeasureSince([]string{"goa", "db", "Query", "list"}, time.Now())
+	var objs []Query
+	err := r.db.Where("space_id = ?", spaceID).Find(&objs).Error
+	if err != nil && err != gorm.ErrRecordNotFound {
+		return nil, err
+	}
+	return objs, nil
+}
+
+// ListByCreator all queries in a space by a creator
+func (r *GormQueryRepository) ListByCreator(ctx context.Context, spaceID uuid.UUID, creatorID uuid.UUID) ([]Query, error) {
+	defer goa.MeasureSince([]string{"goa", "db", "Query", "listbycreator"}, time.Now())
+	var objs []Query
+	err := r.db.Where("space_id = ? AND creator=?", spaceID, creatorID).Find(&objs).Error
+	if err != nil && err != gorm.ErrRecordNotFound {
+		return nil, err
+	}
+	return objs, nil
+}
+
+// Load Query in a space
+func (r *GormQueryRepository) Load(ctx context.Context, ID uuid.UUID, spaceID uuid.UUID) (*Query, error) {
+	defer goa.MeasureSince([]string{"goa", "db", "query", "show"}, time.Now())
+	q := Query{}
+	tx := r.db.Where("id = ? and space_id = ?", ID, spaceID).First(&q)
+	if tx.RecordNotFound() {
+		log.Error(ctx, map[string]interface{}{
+			"query_id": ID.String(),
+		}, "record not found")
+		return nil, errors.NewNotFoundError("query", ID.String())
+	}
+	if tx.Error != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err":      tx.Error,
+			"query_id": ID.String(),
+		}, "unable to load the query by ID")
+		return nil, errors.NewInternalError(ctx, tx.Error)
+	}
+	return &q, nil
+}
+
+// Delete deletes the query with the given id, returns NotFoundError or InternalError
+func (r *GormQueryRepository) Delete(ctx context.Context, ID uuid.UUID) error {
+	defer goa.MeasureSince([]string{"goa", "db", "query", "delete"}, time.Now())
+	q := Query{ID: ID}
+	tx := r.db.Delete(q)
+
+	if err := tx.Error; err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"query_id": ID.String(),
+		}, "unable to delete the query")
+		return errors.NewInternalError(ctx, err)
+	}
+	if tx.RowsAffected == 0 {
+		log.Error(ctx, map[string]interface{}{
+			"query_id": ID.String(),
+		}, "none row was affected by the delete operation")
```

Comment:
typo: "none" should be "no"

Reply:
thanks fixed in https://github.com/fabric8-services/fabric8-wit/pull/1824/commits/354050c56766743806b6eaeb78751e72c7546d95